// app-iris-header/esm5/src/app/core/services/api.service.js
import * as tslib_1 from "tslib";
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { environment } from '../../../environments/environment';
var ApiService = /** @class */ (function () {
    function ApiService(http) {
        this.http = http;
    }
    ApiService.prototype.get = function (path, params) {
        if (params === void 0) { params = new HttpParams(); }
        return this.http.get("" + environment.API_URL + path, { params: params });
    };
    ApiService.prototype.getPagination = function (path, start, end) {
        return this.http.get("" + environment.API_URL + path + "?_start=" + start + "&_end=" + end);
    };
    ApiService.prototype.put = function (path, body) {
        if (body === void 0) { body = {}; }
        return this.http.put("" + environment.API_URL + path, JSON.stringify(body));
    };
    ApiService.prototype.post = function (path, body, httpOptions) {
        if (body === void 0) { body = {}; }
        console.log(JSON.stringify(body));
        return this.http.post("" + environment.API_URL + path, JSON.stringify(body), httpOptions);
    };
    ApiService.prototype.postFormData = function (path, body, httpOptions) {
        if (body === void 0) { body = {}; }
        var newBody = this.getFormUrlEncoded(body);
        return this.http.post("" + environment.API_URL + path, newBody.toString(), httpOptions);
    };
    ApiService.prototype.getFormUrlEncoded = function (toConvert) {
        var urlSearchParams = new URLSearchParams();
        // tslint:disable-next-line:forin
        for (var property in toConvert) {
            var encodedKey = property;
            var encodedValue = toConvert[property];
            urlSearchParams.append(encodedKey, encodedValue);
        }
        return urlSearchParams;
    };
    ApiService.prototype.delete = function (path) {
        return this.http.delete("" + environment.API_URL + path);
    };
    ApiService = tslib_1.__decorate([
        Injectable(),
        tslib_1.__metadata("design:paramtypes", [HttpClient])
    ], ApiService);
    return ApiService;
}());
export { ApiService };
