// app-iris-header/esm2015/src/app/core/services/api.service.js
import * as tslib_1 from "tslib";
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { environment } from '../../../environments/environment';
let ApiService = class ApiService {
    constructor(http) {
        this.http = http;
    }
    get(path, params = new HttpParams()) {
        return this.http.get(`${environment.API_URL}${path}`, { params });
    }
    getPagination(path, start, end) {
        return this.http.get(`${environment.API_URL}${path}?_start=${start}&_end=${end}`);
    }
    put(path, body = {}) {
        return this.http.put(`${environment.API_URL}${path}`, JSON.stringify(body));
    }
    post(path, body = {}, httpOptions) {
        console.log(JSON.stringify(body));
        return this.http.post(`${environment.API_URL}${path}`, JSON.stringify(body), httpOptions);
    }
    postFormData(path, body = {}, httpOptions) {
        const newBody = this.getFormUrlEncoded(body);
        return this.http.post(`${environment.API_URL}${path}`, newBody.toString(), httpOptions);
    }
    getFormUrlEncoded(toConvert) {
        const urlSearchParams = new URLSearchParams();
        // tslint:disable-next-line:forin
        for (const property in toConvert) {
            const encodedKey = property;
            const encodedValue = toConvert[property];
            urlSearchParams.append(encodedKey, encodedValue);
        }
        return urlSearchParams;
    }
    delete(path) {
        return this.http.delete(`${environment.API_URL}${path}`);
    }
};
ApiService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__metadata("design:paramtypes", [HttpClient])
], ApiService);
export { ApiService };
