// app-iris-header/esm5/src/app/shared/components/mini-dropdown/mini-dropdown.component.js
import * as tslib_1 from "tslib";
import { Component, Input, Output, EventEmitter, forwardRef } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
var INPUT_FIELD_VALUE_ACCESSOR = {
    provide: NG_VALUE_ACCESSOR,
    useExisting: forwardRef(function () { return MiniDropdownComponent; }),
    multi: true
};
var MiniDropdownComponent = /** @class */ (function () {
    function MiniDropdownComponent() {
        this.isReadOnly = false;
        this.changeValue = new EventEmitter();
        this.innerValue = '';
        this.onChangeCb = function () { };
        this.onTouchedCb = function () { };
    }
    Object.defineProperty(MiniDropdownComponent.prototype, "value", {
        get: function () {
            return this.innerValue;
        },
        set: function (v) {
            if (v !== this.innerValue) {
                this.innerValue = v;
                this.onChangeCb(v);
            }
        },
        enumerable: true,
        configurable: true
    });
    MiniDropdownComponent.prototype.onBlur = function () {
        this.onTouchedCb;
    };
    MiniDropdownComponent.prototype.writeValue = function (value) {
        if (value !== this.innerValue) {
            this.innerValue = value;
        }
    };
    MiniDropdownComponent.prototype.registerOnChange = function (fn) {
        this.onChangeCb = fn;
    };
    MiniDropdownComponent.prototype.registerOnTouched = function (fn) {
        this.onTouchedCb = fn;
    };
    MiniDropdownComponent.prototype.setDisabledState = function (isDisabled) {
        this.isReadOnly = isDisabled;
    };
    MiniDropdownComponent.prototype.onChange = function (v) {
        this.changeValue.emit(v);
    };
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Array)
    ], MiniDropdownComponent.prototype, "data", void 0);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Object)
    ], MiniDropdownComponent.prototype, "isReadOnly", void 0);
    tslib_1.__decorate([
        Output(),
        tslib_1.__metadata("design:type", Object)
    ], MiniDropdownComponent.prototype, "changeValue", void 0);
    MiniDropdownComponent = tslib_1.__decorate([
        Component({
            selector: 'app-mini-dropdown',
            template: "<select class=\"custom-select d-block w-100 custom-select-iris\" [(ngModel)]=\"value\" (ngModelChange)=\"onChange($event)\">\r\n    <option *ngFor=\"let item of data\" [ngValue]=\"item.id\">{{item.name}}</option>\r\n</select>",
            providers: [INPUT_FIELD_VALUE_ACCESSOR],
            styles: [".custom-select-iris{background-color:#1b7ccb;color:#fff;border-radius:20px;font-size:11px;background-image:url(../../../../assets/img/arrow-down.png);background-size:initial;font-family:Gotham;border:none}"]
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], MiniDropdownComponent);
    return MiniDropdownComponent;
}());
export { MiniDropdownComponent };
