// app-iris-header/bundles/app-iris-header.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@angular/forms'), require('ngx-smart-modal'), require('@ng-bootstrap/ng-bootstrap'), require('@angular/common/http'), require('@angular/common/locales/pt')) :
    typeof define === 'function' && define.amd ? define('app-iris-header', ['exports', '@angular/core', '@angular/common', '@angular/forms', 'ngx-smart-modal', '@ng-bootstrap/ng-bootstrap', '@angular/common/http', '@angular/common/locales/pt'], factory) :
    (global = global || self, factory(global['app-iris-header'] = {}, global.ng.core, global.ng.common, global.ng.forms, global.ngxSmartModal, global.ngBootstrap, global.ng.common.http, global.ng.common.locales.pt));
}(this, function (exports, core, common, forms, ngxSmartModal, ngBootstrap, http, localePt) { 'use strict';

    localePt = localePt && localePt.hasOwnProperty('default') ? localePt['default'] : localePt;

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation. All rights reserved.
    Licensed under the Apache License, Version 2.0 (the "License"); you may not use
    this file except in compliance with the License. You may obtain a copy of the
    License at http://www.apache.org/licenses/LICENSE-2.0

    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
    WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
    MERCHANTABLITY OR NON-INFRINGEMENT.

    See the Apache Version 2.0 License for specific language governing permissions
    and limitations under the License.
    ***************************************************************************** */
    /* global Reflect, Promise */

    var extendStatics = function(d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };

    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }

    var __assign = function() {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };

    function __rest(s, e) {
        var t = {};
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
            t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }

    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }

    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); }
    }

    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(metadataKey, metadataValue);
    }

    function __awaiter(thisArg, _arguments, P, generator) {
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
            function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
            function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }

    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f) throw new TypeError("Generator is already executing.");
            while (_) try {
                if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
                if (y = 0, t) op = [op[0] & 2, t.value];
                switch (op[0]) {
                    case 0: case 1: t = op; break;
                    case 4: _.label++; return { value: op[1], done: false };
                    case 5: _.label++; y = op[1]; op = [0]; continue;
                    case 7: op = _.ops.pop(); _.trys.pop(); continue;
                    default:
                        if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                        if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                        if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                        if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                        if (t[2]) _.ops.pop();
                        _.trys.pop(); continue;
                }
                op = body.call(thisArg, _);
            } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
            if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
        }
    }

    function __exportStar(m, exports) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }

    function __values(o) {
        var m = typeof Symbol === "function" && o[Symbol.iterator], i = 0;
        if (m) return m.call(o);
        return {
            next: function () {
                if (o && i >= o.length) o = void 0;
                return { value: o && o[i++], done: !o };
            }
        };
    }

    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m) return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
        }
        catch (error) { e = { error: error }; }
        finally {
            try {
                if (r && !r.done && (m = i["return"])) m.call(i);
            }
            finally { if (e) throw e.error; }
        }
        return ar;
    }

    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }

    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    };

    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }

    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n]) i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try { step(g[n](v)); } catch (e) { settle(q[0][3], e); } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length) resume(q[0][0], q[0][1]); }
    }

    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }

    function __asyncValues(o) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function(v) { resolve({ value: v, done: d }); }, reject); }
    }

    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
        return cooked;
    };

    function __importStar(mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
        result.default = mod;
        return result;
    }

    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }

    // This file can be replaced during build by using the `fileReplacements` array.
    // `ng build --prod` replaces `environment.ts` with `environment.prod.ts`.
    // The list of file replacements can be found in `angular.json`.
    var environment = {
        production: false,
        defaultLocale: 'pt',
        API_URL: 'http://localhost:3000'
    };
    /*
     * For easier debugging in development mode, you can import the following file
     * to ignore zone related error stack frames such as `zone.run`, `zoneDelegate.invokeTask`.
     *
     * This import should be commented out in production mode because it will have a negative impact
     * on performance if an error is thrown.
     */
    // import 'zone.js/dist/zone-error';  // Included with Angular CLI.

    var ApiService = /** @class */ (function () {
        function ApiService(http) {
            this.http = http;
        }
        ApiService.prototype.get = function (path, params) {
            if (params === void 0) { params = new http.HttpParams(); }
            return this.http.get("" + environment.API_URL + path, { params: params });
        };
        ApiService.prototype.getPagination = function (path, start, end) {
            return this.http.get("" + environment.API_URL + path + "?_start=" + start + "&_end=" + end);
        };
        ApiService.prototype.put = function (path, body) {
            if (body === void 0) { body = {}; }
            return this.http.put("" + environment.API_URL + path, JSON.stringify(body));
        };
        ApiService.prototype.post = function (path, body, httpOptions) {
            if (body === void 0) { body = {}; }
            console.log(JSON.stringify(body));
            return this.http.post("" + environment.API_URL + path, JSON.stringify(body), httpOptions);
        };
        ApiService.prototype.postFormData = function (path, body, httpOptions) {
            if (body === void 0) { body = {}; }
            var newBody = this.getFormUrlEncoded(body);
            return this.http.post("" + environment.API_URL + path, newBody.toString(), httpOptions);
        };
        ApiService.prototype.getFormUrlEncoded = function (toConvert) {
            var urlSearchParams = new URLSearchParams();
            // tslint:disable-next-line:forin
            for (var property in toConvert) {
                var encodedKey = property;
                var encodedValue = toConvert[property];
                urlSearchParams.append(encodedKey, encodedValue);
            }
            return urlSearchParams;
        };
        ApiService.prototype.delete = function (path) {
            return this.http.delete("" + environment.API_URL + path);
        };
        ApiService = __decorate([
            core.Injectable(),
            __metadata("design:paramtypes", [http.HttpClient])
        ], ApiService);
        return ApiService;
    }());

    var NoticeService = /** @class */ (function () {
        function NoticeService(api) {
            this.api = api;
        }
        NoticeService.prototype.get = function () {
            return this.api.get('/aviso');
        };
        NoticeService.prototype.getPagination = function (start, end) {
            return this.api.getPagination('/aviso', start, end);
        };
        NoticeService = __decorate([
            core.Injectable(),
            __metadata("design:paramtypes", [ApiService])
        ], NoticeService);
        return NoticeService;
    }());

    var OccurrenceService = /** @class */ (function () {
        function OccurrenceService(api) {
            this.api = api;
        }
        OccurrenceService.prototype.get = function () {
            return this.api.get("/ocorrencia");
        };
        OccurrenceService.prototype.put = function (occurrence) {
            return this.api.put("/ocorrencia", occurrence);
        };
        OccurrenceService = __decorate([
            core.Injectable(),
            __metadata("design:paramtypes", [ApiService])
        ], OccurrenceService);
        return OccurrenceService;
    }());

    var ProfileService = /** @class */ (function () {
        function ProfileService(api) {
            this.api = api;
        }
        ProfileService.prototype.get = function () {
            return this.api.get("/perfil");
        };
        ProfileService = __decorate([
            core.Injectable(),
            __metadata("design:paramtypes", [ApiService])
        ], ProfileService);
        return ProfileService;
    }());

    var HeaderComponent = /** @class */ (function () {
        function HeaderComponent(noticeService, occurrenceService, profileService, modalService, elementRef) {
            this.noticeService = noticeService;
            this.occurrenceService = occurrenceService;
            this.profileService = profileService;
            this.modalService = modalService;
            this.elementRef = elementRef;
            this.actionSelect = new core.EventEmitter();
            var scriptChat = document.createElement('script');
            scriptChat.type = "text/javascript";
            scriptChat.src = "https://cdn.agentbot.net/core/59ff0a5f9b3b351736d6d5c751b6fc3d.js";
            this.elementRef.nativeElement.appendChild(scriptChat);
        }
        HeaderComponent.prototype.ngOnInit = function () {
            var _this = this;
            this.profileService.get().subscribe(function (data) {
                _this.profiles = data;
                _this.profile = _this.profiles[0].id;
            });
            this.occurrenceService.get().subscribe(function (data) {
                _this.occurrences = data;
                _this.unreadOccurrence = data.filter(function (item) {
                    return !item.lido;
                }).length;
            });
            this.noticeService.get().subscribe(function (data) { return _this.notices = data; });
            setTimeout(function () {
                $aivo.ready(function () {
                    $aivo.user.set("key", "123456");
                    $aivo.user.set("name", "Acaciano Neves Souza");
                    $aivo.user.set("email", "acaciano.neves@gmail.com");
                    $aivo.user.set("device", "11 942480645");
                    $aivo.user.set("instituto", "Instituto Teste");
                });
            }, 1000);
        };
        HeaderComponent.prototype.openModal = function (content) {
            this.modalService.open(content, { ariaLabelledBy: 'modal-basic-title', windowClass: 'iris-modal' }).result.then(function () { }, function (reason) { });
        };
        HeaderComponent.prototype.readMessage = function (occurrence) {
            if (this.unreadOccurrence > 0) {
                occurrence.lido = true;
                this.occurrenceService.put(occurrence);
                this.unreadOccurrence--;
            }
        };
        HeaderComponent.prototype.changeProfile = function () {
            this.actionSelect.emit(this.profile);
        };
        HeaderComponent.prototype.logOut = function () {
            localStorage.removeItem(this.logOutName);
        };
        __decorate([
            core.Input(),
            __metadata("design:type", Object)
        ], HeaderComponent.prototype, "logOutName", void 0);
        __decorate([
            core.Input(),
            __metadata("design:type", Object)
        ], HeaderComponent.prototype, "nameUser", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", Object)
        ], HeaderComponent.prototype, "actionSelect", void 0);
        HeaderComponent = __decorate([
            core.Component({
                selector: 't-systems-iris-header',
                template: "<div class=\"header-iris\">\r\n    <div class=\"header-container\">\r\n        <div class=\"logo\"></div>\r\n        <div class=\"logo-hc\"></div>\r\n        <div class=\"logo-faepa\"></div>\r\n        <div class=\"separator\"></div>\r\n        <ul class=\"nav nav-mtd tabs-info z-depth-1\" role=\"tablist\">\r\n            <li class=\"nav-item nav-btn-iris\">\r\n                <app-mini-dropdown [data]=\"profiles\" [(ngModel)]=\"profile\" (changeValue)=\"changeProfile()\">\r\n                </app-mini-dropdown>\r\n            </li>\r\n            <li class=\"nav-item nav-btn-iris\">\r\n                <app-mini-button (click)=\"openModal(contentOccurrence)\" title=\"Ocorrencias\" icon=\"bell\"\r\n                    [num]=\"unreadOccurrence\"></app-mini-button>\r\n            </li>\r\n            <li class=\"nav-item nav-btn-iris\">\r\n                <app-mini-button (click)=\"openModal(contentNotice)\" icon=\"exclamation-circle\" title=\"Avisos\">\r\n                </app-mini-button>\r\n            </li>\r\n            <div class=\"content-right-header\">\r\n                <div class=\"container-user-login\">\r\n                    Ol\u00E1, {{nameUser}}\r\n                </div>\r\n                <div class=\"container-user-login\">\r\n                    <a (click)=\"logOut()\" href=\"javascript:\">\r\n                        <i class=\"fas fa-cog\" placement=\"bottom\" ngbTooltip=\"Alterar Senha\"\r\n                            tooltipClass=\"tooltip-iris\"></i>\r\n                    </a>\r\n                    |\r\n                    <a href=\"javascript:\">\r\n                        <i class=\"fas fa-comment agent-launcher\" placement=\"bottom\" ngbTooltip=\"Chat\" tooltipClass=\"tooltip-iris\"></i>\r\n                    </a>\r\n                    |\r\n                    <a (click)=\"logOut()\" href=\"javascript:\">\r\n                        <i class=\"fas fa-power-off\" placement=\"bottom\" ngbTooltip=\"Sair\"\r\n                            tooltipClass=\"tooltip-iris\"></i>\r\n                    </a>\r\n                </div>\r\n            </div>\r\n        </ul>\r\n    </div>\r\n</div>\r\n\r\n\r\n\r\n<ng-template #contentOccurrence let-modal>\r\n    <div class=\"modal-header\">\r\n        <h4 class=\"modal-title\" id=\"modal-basic-title\">Ocorr\u00EAncias</h4>\r\n        <button type=\"button\" class=\"close\" aria-label=\"Close\" (click)=\"modal.dismiss('Cross click')\">\r\n            <span aria-hidden=\"true\">&times;</span>\r\n        </button>\r\n    </div>\r\n    <div class=\"modal-body\">\r\n        <div *ngFor=\"let item of occurrences\" style=\"cursor: pointer;\" class=\"alert alert-secondary alert-non-read\"\r\n            [ngClass]=\"{'alert-read': item.lido}\" role=\"alert\">\r\n            <div class=\"date-alert\">\r\n                {{item.data | date: 'dd/MM/yyyy HH:mm'}}\r\n            </div>\r\n            <i class=\"far fa-check-square unread-icon\" *ngIf=\"!item.lido\"></i>\r\n            <i class=\"fas fa-check-square read-icon\" *ngIf=\"item.lido\"></i>\r\n            <a (click)=\"readMessage(item)\">{{item.mensagem}}</a>\r\n        </div>\r\n    </div>\r\n</ng-template>\r\n\r\n<ng-template #contentNotice let-modal>\r\n    <div class=\"modal-header\">\r\n        <h4 class=\"modal-title\" id=\"modal-basic-title\">Avisos</h4>\r\n        <button type=\"button\" class=\"close\" aria-label=\"Close\" (click)=\"modal.dismiss('Cross click')\">\r\n            <span aria-hidden=\"true\">&times;</span>\r\n        </button>\r\n    </div>\r\n    <div class=\"modal-body\">\r\n        <div *ngFor=\"let item of notices\" class=\"alert alert-secondary alert-default-iris\" role=\"alert\">\r\n            {{item.mensagem}} - {{item.data | date: 'dd/MM/yyyy HH:mm'}}\r\n        </div>\r\n    </div>\r\n</ng-template>\r\n\r\n<router-outlet></router-outlet>",
                encapsulation: core.ViewEncapsulation.None,
                styles: [".header-iris{width:100%;height:65px;float:left;box-shadow:0 5px 54px 2px rgba(0,0,0,.31);box-sizing:border-box;background-color:#fff;display:table;padding-top:10px}.header-container{width:1000px;margin:0 auto;position:relative}.logo{width:60px;height:22px;float:left;background:url(../../assets/img/logo.png);margin-top:12px}.logo-hc{width:60px;height:50px;float:left;background:url(../../assets/img/HC.png) 0 0/contain no-repeat;margin-top:4px;margin-left:10px}.logo-faepa{width:40px;height:50px;float:left;background:url(../../assets/img/Faepa.jpg) 0 0/contain no-repeat;margin-top:4px;margin-left:10px}.separator{width:1px;height:50px;float:left;background-color:#eef0f4;margin-left:15px;margin-right:15px}.nav-btn-iris{margin-top:5px;margin-right:5px}.nav-bar-iris{padding:0}.content-right-header{position:absolute;right:0;margin-top:5px}.container-user-login{width:100%;float:left;font-family:Gotham;font-size:11px;font-weight:700;color:#1368ad;text-align:right}.logout-button{margin-left:5px}.iris-modal .modal-content{background-color:#fff;font-family:Gotham;box-shadow:0 2px 28px 3px rgba(0,0,0,.24);border-radius:5px}.light-blue-backdrop{background-color:#5cb3fd}.modal-title{margin-bottom:0;line-height:1.5;font-weight:300;font-size:24px;color:#0870b3}.iris-modal .close{color:#0870b3}.alert-non-read{border-radius:5px;font-size:12px;font-weight:700;background-color:#fff}.alert-read{border-color:#0a88d9;font-weight:400}.date-alert{width:100%;font-size:10px}.unread-icon{position:absolute;right:10px;font-size:16px;top:10px}.read-icon{position:absolute;right:10px;font-size:16px;top:10px;color:#0a88d9}.alert-default-iris{background-color:#fff;font-size:12px}@media screen and (max-width:1000px){.header-container{width:100%;float:left;padding-left:10px;padding-right:10px;box-sizing:border-box}.content-right-header{right:10px}}"]
            }),
            __metadata("design:paramtypes", [NoticeService,
                OccurrenceService,
                ProfileService,
                ngBootstrap.NgbModal,
                core.ElementRef])
        ], HeaderComponent);
        return HeaderComponent;
    }());

    var INPUT_FIELD_VALUE_ACCESSOR = {
        provide: forms.NG_VALUE_ACCESSOR,
        useExisting: core.forwardRef(function () { return MiniDropdownComponent; }),
        multi: true
    };
    var MiniDropdownComponent = /** @class */ (function () {
        function MiniDropdownComponent() {
            this.isReadOnly = false;
            this.changeValue = new core.EventEmitter();
            this.innerValue = '';
            this.onChangeCb = function () { };
            this.onTouchedCb = function () { };
        }
        Object.defineProperty(MiniDropdownComponent.prototype, "value", {
            get: function () {
                return this.innerValue;
            },
            set: function (v) {
                if (v !== this.innerValue) {
                    this.innerValue = v;
                    this.onChangeCb(v);
                }
            },
            enumerable: true,
            configurable: true
        });
        MiniDropdownComponent.prototype.onBlur = function () {
            this.onTouchedCb;
        };
        MiniDropdownComponent.prototype.writeValue = function (value) {
            if (value !== this.innerValue) {
                this.innerValue = value;
            }
        };
        MiniDropdownComponent.prototype.registerOnChange = function (fn) {
            this.onChangeCb = fn;
        };
        MiniDropdownComponent.prototype.registerOnTouched = function (fn) {
            this.onTouchedCb = fn;
        };
        MiniDropdownComponent.prototype.setDisabledState = function (isDisabled) {
            this.isReadOnly = isDisabled;
        };
        MiniDropdownComponent.prototype.onChange = function (v) {
            this.changeValue.emit(v);
        };
        __decorate([
            core.Input(),
            __metadata("design:type", Array)
        ], MiniDropdownComponent.prototype, "data", void 0);
        __decorate([
            core.Input(),
            __metadata("design:type", Object)
        ], MiniDropdownComponent.prototype, "isReadOnly", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", Object)
        ], MiniDropdownComponent.prototype, "changeValue", void 0);
        MiniDropdownComponent = __decorate([
            core.Component({
                selector: 'app-mini-dropdown',
                template: "<select class=\"custom-select d-block w-100 custom-select-iris\" [(ngModel)]=\"value\" (ngModelChange)=\"onChange($event)\">\r\n    <option *ngFor=\"let item of data\" [ngValue]=\"item.id\">{{item.name}}</option>\r\n</select>",
                providers: [INPUT_FIELD_VALUE_ACCESSOR],
                styles: [".custom-select-iris{background-color:#1b7ccb;color:#fff;border-radius:20px;font-size:11px;background-image:url(../../../../assets/img/arrow-down.png);background-size:initial;font-family:Gotham;border:none}"]
            }),
            __metadata("design:paramtypes", [])
        ], MiniDropdownComponent);
        return MiniDropdownComponent;
    }());

    var MiniButtonComponent = /** @class */ (function () {
        function MiniButtonComponent() {
        }
        MiniButtonComponent.prototype.ngOnInit = function () {
        };
        __decorate([
            core.Input(),
            __metadata("design:type", String)
        ], MiniButtonComponent.prototype, "title", void 0);
        __decorate([
            core.Input(),
            __metadata("design:type", String)
        ], MiniButtonComponent.prototype, "num", void 0);
        __decorate([
            core.Input(),
            __metadata("design:type", String)
        ], MiniButtonComponent.prototype, "icon", void 0);
        MiniButtonComponent = __decorate([
            core.Component({
                selector: 'app-mini-button',
                template: "<button type=\"button\" class=\"btn btn-primary btn-iris\">\r\n    <i class=\"fas fa-{{icon}} icon-button-iris\"></i>\r\n    {{title}} <span class=\"badge badge-light\">{{num}}</span>\r\n</button>",
                styles: [".btn-iris{background-color:#e6e6e6;border:none;font-size:10px;font-family:Gotham;border-radius:20px;padding:8px 8px 8px 25px;color:#185689;font-weight:700;text-transform:uppercase;position:relative}.icon-button-iris{position:absolute;left:10px;top:10px}"]
            }),
            __metadata("design:paramtypes", [])
        ], MiniButtonComponent);
        return MiniButtonComponent;
    }());

    common.registerLocaleData(localePt);
    var HeaderModule = /** @class */ (function () {
        function HeaderModule() {
        }
        HeaderModule = __decorate([
            core.NgModule({
                declarations: [
                    HeaderComponent,
                    MiniDropdownComponent,
                    MiniButtonComponent
                ],
                imports: [
                    common.CommonModule,
                    forms.FormsModule,
                    ngxSmartModal.NgxSmartModalModule.forRoot(),
                    ngBootstrap.NgbModule
                ],
                exports: [
                    HeaderComponent,
                    MiniButtonComponent
                ],
                schemas: [core.CUSTOM_ELEMENTS_SCHEMA],
                providers: [
                    ApiService,
                    NoticeService,
                    OccurrenceService,
                    ProfileService
                ],
            })
        ], HeaderModule);
        return HeaderModule;
    }());

    exports.HeaderModule = HeaderModule;
    exports.ɵa = HeaderComponent;
    exports.ɵb = NoticeService;
    exports.ɵc = ApiService;
    exports.ɵd = OccurrenceService;
    exports.ɵe = ProfileService;
    exports.ɵf = MiniDropdownComponent;
    exports.ɵg = MiniButtonComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
