// app-iris-header/esm5/src/app/header/header.module.js
import * as tslib_1 from "tslib";
import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { CommonModule, registerLocaleData } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgxSmartModalModule } from 'ngx-smart-modal';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { ApiService } from '../core/services/api.service';
import { NoticeService } from '../core/services/notice.service';
import { OccurrenceService } from '../core/services/occurrence.service';
import { ProfileService } from '../core/services/profile.service';
import localePt from '@angular/common/locales/pt';
import { HeaderComponent } from './header.component';
import { MiniDropdownComponent } from '../shared/components/mini-dropdown/mini-dropdown.component';
import { MiniButtonComponent } from '../shared/components/mini-button/mini-button.component';
registerLocaleData(localePt);
var HeaderModule = /** @class */ (function () {
    function HeaderModule() {
    }
    HeaderModule = tslib_1.__decorate([
        NgModule({
            declarations: [
                HeaderComponent,
                MiniDropdownComponent,
                MiniButtonComponent
            ],
            imports: [
                CommonModule,
                FormsModule,
                NgxSmartModalModule.forRoot(),
                NgbModule
            ],
            exports: [
                HeaderComponent,
                MiniButtonComponent
            ],
            schemas: [CUSTOM_ELEMENTS_SCHEMA],
            providers: [
                ApiService,
                NoticeService,
                OccurrenceService,
                ProfileService
            ],
        })
    ], HeaderModule);
    return HeaderModule;
}());
export { HeaderModule };
